Signal/src/ViewControllers/AppSettingsViewController.m

```diff
@@ -374,12 +374,24 @@ - (void)unregisterUser
 
 - (void)proceedToUnregistration
 {
-    [TSAccountManager unregisterTextSecureWithSuccess:^{
-        [Environment resetAppData];
-    }
-        failure:^(NSError *error) {
-            [OWSAlerts showAlertWithTitle:NSLocalizedString(@"UNREGISTER_SIGNAL_FAIL", @"")];
-        }];
+    [ModalActivityIndicatorViewController
```

Comment:
The downside of the modal's API is that you end up with JS-style "callback hell."

Reply:
Proposal for a little readability improvement...

As I understand it, this modal is used to block the user from interacting with the app while a (relatively) long running background task occurs. As such, I think it would be reasonable to have the block passed execute on a background queue by default - saving the caller from having to dispatch.

In that case I'd also probably rename the param for clarity:

```
[ModalActivityIndicatorViewController presentFromViewController:self
                                                      canCancel:NO
                                                backgroundBlock:^(ModalActivityIndicatorViewController *modalActivityIndicator) {

    [TSAccountManager unregisterTextSecureWithSuccess:^{
                             [Environment resetAppData];
                         }
                                               failure:^(NSError *error) {
                                 dispatch_async(dispatch_get_main_queue(), ^{
                                     [modalActivityIndicator dismissWithCompletion:^{
                                         [OWSAlerts
                                             showAlertWithTitle:NSLocalizedString(@"UNREGISTER_SIGNAL_FAIL", @"")];
                                     }];
                                 });
                             }];
}];
```

Reply:
Done. https://github.com/WhisperSystems/Signal-iOS/pull/2548